xcodeproj/internal/opts.bzl

```diff
@@ -170,11 +171,16 @@ def _get_unprocessed_compiler_opts(
                 ctx.disabled_features + _UNSUPPORTED_CC_FEATURES
             ),
         )
+        include_directories = depset(
+            compilation_context.includes.to_list(), 
+            transitive = [info.compilation_providers.cc_info.compilation_context.includes for attr, info in transitive_infos if info.compilation_providers.cc_info]
+        )
```

Comment:
There are a set of hacky and unsafe assumptions in these lines.

Reply:
If instead y'all add `local_includes` to https://bazel.build/rules/lib/CompilationContext, similar to https://bazel.build/rules/lib/CompilationContext#local_defines, then we could use that instead of `compilation_context.includes`, and it would work for both normal and implementation deps.

Reply:
@ivan-golub

Reply:
Good point, let me check on that option!

Reply:
thx for suggestion @brentleyjones. This is an option for sure but Im wondering how rules_xcodeproj are handling cc_library implementation deps now? Given that we use same approach for objc_library implementaiton deps, so I would assume we can use same approach instead of introducing new concept for compilation context. Main reason here is that Im actually not 100% sure any of objc impl deps changes will be accepted upstream before objc rules are extracted to starlark by bazel core team and changing base compilation contexts reduced that possibility even further

Reply:
We might not be handling them correctly, I haven't checked. And if we aren't, then we would have to do something similar to this (or something similar to Swift, where we grab the arguments from the produced action... but I would really prefer not to go there), which is annoying since `cc_complication_context` currently has all of the information needed to compile the target.

Reply:
Looking into the source code, yeah, we probably don't support `cc_library.implementation_deps` either. I'll take a stab at a version of this change.

Reply:
We need to do more than includes (like defines as well), so I think what we need to do here is call `cc_common.merge_cc_infos`. Seems we already do that, in one case at least, in `compilation_providers.merge`. I'll adjust that code and pipe it through to fix this.

Reply:
https://github.com/MobileNativeFoundation/rules_xcodeproj/pull/1967 should fix it.